extensions/dicom-sr/src/OHIFCornerstoneSRViewport.js

```diff
@@ -81,25 +92,89 @@ function OHIFCornerstoneSRViewport({
     ] = useTrackedMeasurements();
   }
 
+  // Locked if tracking any series
+  let isLocked = trackedMeasurements?.context?.trackedSeries?.length > 0;
+  useEffect(() => {
+    isLocked = trackedMeasurements?.context?.trackedSeries?.length > 0;
+  }, [trackedMeasurements]);
+
+  function _getToolAlias() {
+    const primaryToolId = ToolBarService.state.primaryToolId;
+    let toolAlias = primaryToolId;
+
+    switch (primaryToolId) {
+      case 'Length':
+        toolAlias = 'SRLength';
+        break;
+      case 'Bidirectional':
+        toolAlias = 'SRBidirectional';
+        break;
+      case 'ArrowAnnotate':
+        toolAlias = 'SRArrowAnnotate';
+        break;
+      case 'EllipticalRoi':
+        toolAlias = 'SREllipticalRoi';
+        break;
+    }
+
+    return toolAlias;
+  }
+
   const onElementEnabled = evt => {
     const eventData = evt.detail;
     const targetElement = eventData.element;
+    const toolAlias = _getToolAlias(); // These are 1:1 for built-in only
 
-    // TODO -> This will only be temporary until we set a tool on, and isn't very customizable.
-    // Need to discuss how to deal with tools in general in the redesign, since we
-    // Previously just had Tool mode state global across the entire viewer.
-    const globalTools = cornerstoneTools.store.state.globalTools;
-    const globalToolNames = Object.keys(globalTools);
-
-    globalToolNames.forEach(globalToolName => {
-      cornerstoneTools.setToolDisabledForElement(targetElement, globalToolName);
-    });
-
+    // ~~ MAGIC
+    cornerstoneTools.addToolForElement(targetElement, DICOMSRDisplayTool);
     cornerstoneTools.setToolEnabledForElement(
       targetElement,
       TOOL_NAMES.DICOM_SR_DISPLAY_TOOL
     );
 
+    // ~~ Variants
+    cornerstoneTools.addToolForElement(
```

Comment:
We need the angle and the rectangle here, they currently are still messed up like before.